java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -7,462 +7,548 @@
+        BigDecimal westLongitude,
+        BigDecimal northLatitude,
+        BigDecimal eastLongitude) {
+      this.southLatitude = southLatitude;
+      this.westLongitude = westLongitude;
+      this.northLatitude = northLatitude;
+      this.eastLongitude = eastLongitude;
     }
-    private static final char SEPARATOR = '+';
-    private static final char SEPARATOR_POSITION = 8;
-    private static final char SUFFIX_PADDING = '0';
-
-    /** Class providing information about area covered by Open Location Code. */
-    public class CodeArea {
-        private final BigDecimal southLatitude;
-        private final BigDecimal westLongitude;
-        private final BigDecimal northLatitude;
-        private final BigDecimal eastLongitude;
-
-        public CodeArea(BigDecimal southLatitude, BigDecimal westLongitude, BigDecimal northLatitude, BigDecimal eastLongitude) {
-            this.southLatitude = southLatitude;
-            this.westLongitude = westLongitude;
-            this.northLatitude = northLatitude;
-            this.eastLongitude = eastLongitude;
-        }
 
-        public double getSouthLatitude() {
-            return southLatitude.doubleValue();
-        }
+    public double getSouthLatitude() {
+      return southLatitude.doubleValue();
+    }
 
-        public double getWestLongitude() {
-            return westLongitude.doubleValue();
-        }
+    public double getWestLongitude() {
+      return westLongitude.doubleValue();
+    }
 
-        public double getLatitudeHeight() {
-            return northLatitude.subtract(southLatitude).doubleValue();
-        }
+    public double getLatitudeHeight() {
+      return northLatitude.subtract(southLatitude).doubleValue();
+    }
 
-        public double getLongitudeWidth() {
-            return eastLongitude.subtract(westLongitude).doubleValue();
-        }
+    public double getLongitudeWidth() {
+      return eastLongitude.subtract(westLongitude).doubleValue();
+    }
 
-        public double getCenterLatitude() {
-            return southLatitude.add(northLatitude).doubleValue() / 2;
-        }
+    public double getCenterLatitude() {
+      return southLatitude.add(northLatitude).doubleValue() / 2;
+    }
 
-        public double getCenterLongitude() {
-            return westLongitude.add(eastLongitude).doubleValue() / 2;
-        }
+    public double getCenterLongitude() {
+      return westLongitude.add(eastLongitude).doubleValue() / 2;
+    }
 
-        public double getNorthLatitude() {
-            return northLatitude.doubleValue();
-        }
+    public double getNorthLatitude() {
+      return northLatitude.doubleValue();
+    }
 
-        public double getEastLongitude() {
-            return eastLongitude.doubleValue();
-        }
+    public double getEastLongitude() {
+      return eastLongitude.doubleValue();
     }
+  }
 
 
-    /** The state of the OpenLocationCode. */
-    private final String code;
+  /**
+   * The state of the OpenLocationCode.
+   */
```

Comment:
FYI, the one-line javadoc comments are also allowed, and for one line definition probably more readable.